### PR TITLE
fix: fix miss case for collect unflush data file

### PR DESCRIPTION
**Pull request by ZENOTME**
This PR:
- fix base file writer
- refine iceberg writer trait

### CODE REVIEW

File: icelake/src/io_v2/file_writer/base_file_writer.rs [hunk 1]
```diff
@@ -276,20 +272,40 @@ mod prometheus {
...
-            self.inner.close().await
+            let res = self.inner.close().await?;
+            let delta = (res.len() - self.cur_metrics.unflush_data_file) as i64;
+            assert!(delta >= 0);
+            self.unflush_data_file.add(delta as u64);
+            Ok(res)
+        }
+    }
+
+    impl<B: FileWriterBuilder> SingleFileWriterStatus for BaseFileWriterWithMetrics<B>
```
Comment (liurenjie1024) [inline icelake/src/io_v2/file_writer/base_file_writer.rs:295]: This seems interesting. If rolling file writer also needs this metrics, how about make it part of `FileWriter`?

Comment (ZENOTME) [reply icelake/src/io_v2/file_writer/base_file_writer.rs:295]: Yes, make it part of FileWriter is more simple. But we still need it in IcebergWriter. How about 
```
#[async_trait::async_trait]
pub trait FileWriter: Send + 'static + SingleFileWriterStatus {
    type R: FileWriteResult;
    async fn write(&mut self, batch: &RecordBatch) -> Result<()>;
    async fn close(self) -> Result<Vec<Self::R>>;
}
```
The benefit of it is that we don't need to maintain the duplicate interface: one in FileWriter and one for the SingleFileWriterStatus trait.🤔 And we can guarantee that all the FileWriter implement this interface.

File: icelake/src/io_v2/base_writer/position_delete_writer.rs [hunk 2]
```diff
@@ -32,7 +32,9 @@ impl<B: FileWriterBuilder> PositionDeleteWriterBuilder<B> {
 }
 
 #[async_trait::async_trait]
-impl<B: FileWriterBuilder> IcebergWriterBuilder for PositionDeleteWriterBuilder<B> {
+impl<B: FileWriterBuilder> IcebergWriterBuilder<PositionDeleteInput>
```
[the remark this answers is not in the document]
Comment (ZENOTME) [reply icelake/src/io_v2/base_writer/position_delete_writer.rs:35]: https://github.com/icelake-io/icelake/blob/134ca92500ecd880824caa356a8a08b174cac11c/icelake/src/io_v2/base_writer/position_delete_writer.rs#L97

File: icelake/src/io_v2/file_writer/base_file_writer.rs (hunk 1 above)
Comment (liurenjie1024) [reply icelake/src/io_v2/file_writer/base_file_writer.rs:295]: LGTM. How about renaming it to `CurrentFileStatus`? It sounds more clear.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**